### ISSUE TITLE
Use application specific data kind 30078 (NIP-78)

### ISSUE THREAD

**Issue by nobu-maeda** [MostroP2P/mostro#50]
As titled. [NIP-78](https://github.com/nostr-protocol/nips/blob/master/78.md). This can open up `content` and `tag` field to adopt application specific meanings. See https://github.com/nobu-maeda/n3xb-nostr-derisk as an example.

**Comment by grunch** [comment]
our events are published with a d tag already, in the d tag we have the order id, this allow us to modify the event in the future, could add a readme on the `n3xb-nostr-derisk` repo explaining how to use and what it does?

**Comment by grunch** [comment]
I'm closing this issue due inactivity, feel free to open it up again if you want to bring more details about it

**Comment by nobu-maeda** [comment]
> our events are published with a d tag already, in the d tag we have the order id, this allow us to modify the event in the future, could add a readme on the `n3xb-nostr-derisk` repo explaining how to use and what it does?

I think this looks not relevant now because there's the assumption on a single instance of Mostro posting all offers on behalf of makers? I got this from [here](https://github.com/MostroP2P/mostro-cli/blob/df5158ae5ca2b257600af5fc361879aabdf528b1/src/cli.rs#L167). A client only ever queries for offers authored by a single pubkey at a time

Assuming a world where there are many Mostro coordinator instances creating offers. Or better yet, the coordinator-less directly offer publishing that I talked about [here](https://github.com/MostroP2P/mostro/discussions/55), then the assumption to query based on pubkey will no longer be possible. This is when a tag that denotes a Nostr event being a Mostro note (or all n3xB compatible notes) will be important. Other tags can be created like Buy vs Sell offer, amongst other filterable categoric parameters. This will reduce the amount of notes returned back to the client - without any smarts baked into relays through more NIPs. All that's needed is to take full advantage of NIP-78 and NIP-12.
